app/models/notification_config.rb

```diff
@@ -0,0 +1,51 @@
+class NotificationConfig < ApplicationRecord
+  extend Enumerize
+
+  belongs_to :project
+
+  enumerize :delivery_type, in: [:email, :pager_duty, :zulip]
+
+  validates_presence_of :delivery_type
+  validate :params_match_delivery_type
+
+  def handle_ticket(ticket)
+    return unless filter_match?(ticket)
+
+    case delivery_type
+    when 'email'
+      logger.debug 'Sending email'
+    when 'pager_duty'
+      PagerDutySender.handle_ticket_notification(ticket, self)
+    when 'zulip'
+      ZulipSender.handle_ticket_notification(ticket, self)
+    end
```

Comment:
When you use enumerize that can be done better. I'll push a commit later illustrating that